src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private Node<K,V>[] table;
+    private int size;
+
+    private class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        setInitialCapacity();
+        Node<K, V> node = new Node<>(key, value, null);
+        int index = getIndex(node.key);
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        } else {
+            Node<K, V> newNode = table[index];
+            while (newNode != null) {
+                if (Objects.equals(node.key, newNode.key)) {
+                    newNode.value = value;
+                    return;
+                }
+                if (newNode.next == null) {
+                    newNode.next = node;
+                    size++;
+                    return;
+                }
+                newNode = newNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
```

Comment:
Instead of checking if table is null, it's better to initialize the table in the constructor. This way, you can avoid unnecessary null checks in the getValue method.

Reply:
This check should be removed, the key can be as null